### PR TITLE
refactor(video): move encoder declarations to header

### PR DESCRIPTION
**Pull request by ReenigneArcher**
## Description

This PR aims to simplify creation of encoder fixtures for unit testing (#1603).

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I want maintainers to keep my branch updated

### CODE REVIEW

**Comment by ReenigneArcher** [comment]
Docs build fails due to duplicate C++ declaration. https://github.com/breathe-doc/breathe/issues/972

File: src/video.h [hunk 1]
```diff
@@ -11,11 +11,183 @@
...
+      std::string name;
+      std::variant<int, int *, std::optional<int> *, std::function<int()>, std::string, std::string *> value;
+
+      option_t(std::string &&name, decltype(value) &&value):
+          name { std::move(name) }, value { std::move(value) } {}
+    };
+
+    const std::unique_ptr<const encoder_platform_formats_t> platform_formats;
+
+    struct profile {
```
Comment (ReenigneArcher) [inline src/video.h:127]: Note, this was `struct {` before. Changing this allowed the docs to build without the `Duplicate C++ Declaration` warning.

Comment (kentyman23) [reply src/video.h:127]: It's good practice to give all types --- especially complex ones --- clear names. And if it was hard to name, that's a code smell that it doesn't have a cohesive, single responsibility.

**Review by kentyman23 — APPROVED** [review]
My `using` directive comment is the only one that might warrant a change.

File: src/video.h [hunk 2]
```diff
@@ -11,11 +11,181 @@
...
+        _CONVERT(REF_FRAMES_RESTRICT);
+        _CONVERT(CBR);
+        _CONVERT(DYNAMIC_RANGE);
+        _CONVERT(VUI_PARAMETERS);
+        _CONVERT(MAX_FLAGS);
+      }
+#undef _CONVERT
+
+      return {"unknown"};
+    }
```
Comment (ReenigneArcher) [inline src/video.h:110]: This was refactored slightly to avoid `using namespace std::literal;` in the header.

File: src/video.cpp [hunk 3]
```diff
@@ -14,7 +14,6 @@ extern "C" {
 #include <libavutil/mastering_display_metadata.h>
 #include <libavutil/opt.h>
 #include <libavutil/pixdesc.h>
-#include <libswscale/swscale.h>
```
Comment (cgutman) [inline src/video.cpp:17]: Is something else pulling this in for the swscale APIs below?

Comment (ReenigneArcher) [reply src/video.cpp:17]: Probably swscale from the video.h

Comment (ReenigneArcher) [reply src/video.cpp:17]: I can leave it in `video.cpp` as well, if that's preferred.

**Comment by ReenigneArcher** [comment]
Converted to draft as there are a probably a few declarations that would need to be moved to the header in order to use for unit testing.

At least the following, but maybe more.

```cpp
  bool
  validate_encoder(encoder_t &encoder, bool expect_failure);
```

**Comment by ReenigneArcher** [comment]
I have a working H264 software encoder fixture, so I think this is ready

**Pull request merged**